decidim-generators/lib/decidim/generators/app_templates/ai_initializer.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+if Decidim.module_installed?(:ai)
+  Decidim::Ai::Language.service = "Decidim::Ai::Language::Detection"
+  Decidim::Ai::Language.formatter = "Decidim::Ai::Language::Formatter"
+
+  Decidim::Ai::SpamDetection.reporting_user_email = "your-admin@example.org"
+
+  Decidim::Ai::SpamDetection.resource_score_threshold = 0.75 # default
+
+  # The entry must be a hash with the following keys:
+  # - name: the name of the analyzer
+  # - strategy: the class of the strategy to use
+  # - options: a hash with the options to pass to the strategy
+  # Example:
+  # Decidim::Ai::SpamDetection.resource_analyzers = [
+  #   {
+  #     name: :bayes,
+  #     strategy: Decidim::Ai::SpamContent::BayesStrategy,
+  #     options: {
+  #       adapter: :redis,
+  #       params: {
+  #         url:                lambda { ENV["REDIS_URL"] }
+  #         scheme:             "redis"
+  #         host:               "127.0.0.1"
+  #         port:               6379
+  #         path:               nil
+  #         timeout:            5.0
+  #         password:           nil
+  #         db:                 0
+  #         driver:             nil
+  #         id:                 nil
+  #         tcp_keepalive:      0
+  #         reconnect_attempts: 1
+  #         inherit_socket:     false
+  #       }
+  #     }
+  #   }
+  # ]
+  Decidim::Ai::SpamDetection.resource_analyzers = [
+    {
+      name: :bayes,
+      strategy: Decidim::Ai::SpamDetection::Strategy::Bayes,
+      options: {
+        adapter: ENV.fetch("DECIDIM_SPAM_DETECTION_BACKEND_RESOURCE", "redis"),
+        params: { url: ENV.fetch("DECIDIM_SPAM_DETECTION_BACKEND_RESOURCE_URL", {}) }
```

Comment:
We should do the same as we do for ActionCable:

1. Use the REDIS_URL env var
2. Fallback to `redis://localhost:6379/1`

https://github.com/decidim/decidim/blob/2b64bc3d315123a9826870000d60a6d13db59a34/decidim-generators/lib/decidim/generators/app_templates/cable.yml.erb#L9

Reply:
To be honest, i do not want to mix things between spam and action cable or sidekiq. I am not 100% percent how the data is being stored.

Reply:
Ok, you're right that this may be a problem. Then I propose the following: 

1. To keep it separate
2. To improve the naming so it's clear
3. To use another DB as default instead of "/1" that's used by ActionCable. Redis has 16 databases enabled by default

```diff
-params: { url: ENV.fetch("DECIDIM_SPAM_DETECTION_BACKEND_RESOURCE_URL", {}) }
+params: { url: ENV.fetch("DECIDIM_AI_SPAM_DETECTION_BACKEND_RESOURCE_REDIS_URL", { "redis://localhost:6379/2" }) }
```